### PR TITLE
Admindashboard

### PR DESCRIPTION
**Pull request by MahmoudMH**

### CODE REVIEW

File: src/client/actions/viewStudentsAction.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+export const studentsFetchFailure = massage => {
+
+  return {
+    type: STUDENTS_FETCH_FAILURE,
+    error: massage
+  };
+};
+
+export const fetchStudentsView = () => dispatch => {
+  console.log('it is me ^_^');
```
Comment (astroash) [inline src/client/actions/viewStudentsAction.js]: can we remove this console.log, or atleast have a more descriptive message 😊

File: src/client/components/admindashborad/filter/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react';
+
+import './style.css';
+
+class Filter extends Component {
```
Comment (astroash) [inline src/client/components/admindashborad/filter/index.js]: If this component has no state or lifecycle methods, lets make it a [functional component](https://reactjs.org/docs/components-and-props.html#functional-and-class-components).

File: src/client/components/admindashborad/index.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+class Admindashboard extends Component {
+  componentDidMount() {
+    const { fetchStudentsView } = this.props;
+    fetchStudentsView();
+
+  }
+  render() {
+    const { isFetching, error , dataStudents } = this.props.data;
+    // const dataStudents = this.props.data.dataStudents;
```
Comment (astroash) [inline src/client/components/admindashborad/index.js]: 🔥Remove commented out code

File: src/client/components/admindashborad/index.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    console.log(dataStudents,'aaaaaaaaaaa');
```
Comment (astroash) [inline src/client/components/admindashborad/index.js]: 🔥 Remove console.logs

File: src/client/components/admindashborad/index.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    console.log(isFetching,' isFetching');
+
+    return (
+
+      <div className='admindashboard'>
+        <div className='testNav'></div>
+        <div className='admindashboard__container'>
```
Comment (astroash) [inline src/client/components/admindashborad/index.js]: Great css class naming conventions 👌🏽

File: src/client/components/admindashborad/students/style.css [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  align-items: center;
+  height: 140px;
+  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.25);
+  width: 540px;
+  padding: 10px;
+  justify-content: space-between;
+  background-color: #fff;
+  margin: 10px 0;
+}
+.student__info{
```
Comment (astroash) [inline src/client/components/admindashborad/students/style.css]: Do you need this empty class?

File: src/client/index.js [hunk 7]
```diff
@@ -1,18 +1,27 @@
...
 import ReactDOM from 'react-dom';
-import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
+
+import { Provider } from 'react-redux';
+import store from './store';
+
 import Home from './components/home';
 import Page404 from './components/page404';
+import Admindashborad from './components/admindashborad';
```
Comment (astroash) [inline src/client/index.js]: misspelled variable name. Also recommend [pascal case](http://wiki.c2.com/?PascalCase) for component names eg `AdminDashboard`

File: src/client/index.js [hunk 8]
```diff
@@ -1,18 +1,27 @@
...
-        <Route component={Page404} />
-      </Switch>
-    </div>
-  </BrowserRouter>,
+  <Provider store={store}>
+    <BrowserRouter>
+      <div className='app'>
+        <Switch>
+          <Route exact path='/' component={Home} />
+          <Route exact path='/dashborad' component={Admindashborad} />
```
Comment (astroash) [inline src/client/index.js]: `/dashborad` -> `/dashboard`

File: src/client/components/admindashborad/students/index.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+            <div className='student__name'>{this.props.dataStudent.username}</div>
+            <div className='student__job student__span'>Full Stack Developer</div>
+            <div><span className='student__span'>Skills: </span>
+              {
+                arrayOfSkills.map((skill,i) => {
+                  if (i !== arrayOfSkills.length-1) {
+                    return `${skill}, `;
+                  } else {
+                    return `${skill}`;
+                  }
```
[the remark this answers is not in the document]
Comment (yahyaHB) [reply src/client/components/admindashborad/students/index.js]: and I think you should change the whole way of the map

File: src/client/actions/viewStudentsAction.js (hunk 1 above)
Comment (MahmoudMH) [reply src/client/actions/viewStudentsAction.js]: ok that's done @astroash

File: src/client/components/admindashborad/filter/index.js (hunk 2 above)
Comment (MahmoudMH) [reply src/client/components/admindashborad/filter/index.js]: it's stateful component but i was not finished

File: src/client/components/admindashborad/index.js (hunk 3 above)
Comment (MahmoudMH) [reply src/client/components/admindashborad/index.js]: done

File: src/client/components/admindashborad/index.js (hunk 4 above)
Comment (MahmoudMH) [reply src/client/components/admindashborad/index.js]: done

File: src/client/components/admindashborad/students/style.css (hunk 6 above)
Comment (MahmoudMH) [reply src/client/components/admindashborad/students/style.css]: yes i don't need it and i removed it

File: src/client/components/admindashborad/students/index.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (MahmoudMH) [reply src/client/components/admindashborad/students/index.js]: at line 25 there is comma and white space that and else if the it's last element i will not add space and comma @yahyaHB

File: src/client/components/admindashborad/index.js (hunk 5 above)
Comment (MahmoudMH) [reply src/client/components/admindashborad/index.js]: it should remove lately, because i wanna Yahya finished it

**Pull request closed without merging**